packages/wallets/src/evm/wallets/base.ts

```diff
@@ -48,15 +48,6 @@ export abstract class AbstractBrowserWallet<
       return;
     }
 
-    const connector = await this.getConnector();
-    const isConnected = await connector.isConnected();
-
-    // return if already connected
-    if (isConnected) {
-      this.#subscribeToEvents(connector);
-      return;
-    }
```

Comment:
Why remove this code from here?

Reply:
Because of this, the wallet.addEventListener() does not work

Reply:
when auto-connecting a wallet, `isConnected` is `true` and because of that the call to `this.connect(...)` is skipped - but it shouldn't be. because of this connector's `connect` method is not called when auto-connecting and because of that wallet does not setup event listeners for the provider ( which is done in connector.connect method )

<img width="460" alt="image" src="https://user-images.githubusercontent.com/22043396/223654122-1b0a5f69-cb4e-4082-b61e-8107eb78d218.png">

Reply:
checking the `connector.isConnected()` ( here the connecotr is wagmi adapter)  internally forwards it to `connector.isAuthorized()` 

<img width="466" alt="image" src="https://user-images.githubusercontent.com/22043396/223655989-ea8903ea-55a7-4674-9b54-f66b3d9504e5.png">

Here, it checks for storage, provider and provider address and return true/false based on that. In case of autoconnect - we get the provider address and so it returns `true` - but these ends up skipping the steps required in `connect` method mentioned in previous comment.

<img width="733" alt="image" src="https://user-images.githubusercontent.com/22043396/223655956-bfbba13e-2e60-469f-a716-4a6f4389151b.png">

So to fix this, I removed the condition and we always use the "connect flow" for autoconnect instead of using a short cut